### ISSUE TITLE
Rework of Spot-Perpetual Arbitrage logic

### ISSUE THREAD

**Issue by phbrgnomo** [CoinAlpha/hummingbot#3791]
## Why

The trading logic of the spot-perpetual strategy needs improvement. As mentioned on #3679, the current logic of the spot-perpetual arbitrage can result in inefficient trade operations being executed.

The strategy currently allows the trade operation to start with a negative price divergence, initiating the trade at a loss, with the expectation of closing the trade on an opposing divergence.

This trading logic isn't optimal, since it adds more risk to the operation because it usually increases the time the trader is positioned until a profitable exit emerges, increasing the risk of liquidation.

## What

Rework the strategy logic so it does not open a position at a loss, hoping it would go up to make a profit. And implement the following suggestions for improvement based on @Nullably's code review:

- Rename `min_divergence` config to `min_opening_arbitrage_pct` to clearly describe what the config does (to open an arbitrage position)
- Rename `min_convergence` to `min_closing_arbitrage_pct`
- Rename `next_arbitrage_cycle_delay` to `next_arbitrage_opening_delay`
- Rename and replace `derivative` with `perpetual` to clearly specify the type of derivative trading
- Arbitrage pct for both `min_opening_arbitrage_pct` and `min_closing_arbitrage_pct` is calculated as (sell price - buy price) / buy_price, thus negative value indicates negative profit expectation.
- `min_opening_arbitrage_pct` should always be positive value in real trading, it also allows negative value for testing purposes (like min_profitability in other arbitrage strategies)
- `min_closing_arbitrage_pct` can be negative value if the user wants to close the position at lesser profit but at a higher chance of closing.
- The `ArbProposal` should be a simple data class (one buy side and one sell side), there should not be any operation on the connectors here, no calling `get_order_price` here, those prices should be supplied at init
- The strategy should generate a list of all possible ArbProposal and use them as appropriate and the list should be refreshed on every tick (or simply not stored as a strategy object member).
- The conditions for opening arb position and closing it should be clear, separated and easy to follow in the code
- Try not to store calculated values as class members, this can lead to uncertainty on the integrity of the values  
- `first arbitrage` and `second arbitrage` on the messages are unclear here, they are for the same arbitrage position, one is to open and the other is to close it
- Remove `max_funding_rate` configuration parameter for now to get the basics of the strategy right first

## Definition of done
- [ ] Config parameters change as described
- [ ] The strategy first checks if the user has any balances enough at both markets to start the strategy, if not the bot logs balance insufficient message.
- [ ] When an arbitrage opening is found (arbitrage pct > `min_opening_arbitrage_pct`), the bot submits orders and logs messages correctly.
- [ ] When an arbitrage closing is found (arbitrage pct > `min_closing_arbitrage_pct`), the bot submits orders to close the position and logs messages correctly.
- [ ] If there is an existing position on perpetual before the start of the strategy, there are 2 scenarios.
  - If the position amount matches strategy `order_amount`, the strategy assumes the first leg of the arbitrage is done and continues with the second leg operation. The strategy logs proper messages.
  - If the position amount doesn't match, the strategy doesn't continue and it logs message to prompt user to manually close the position before starting the strategy.
- [ ] Add / create unit tests for this strategy

**Comment by dennisocana** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/hummingbot-5f340755c5e17300167a8f30/issues/CoinAlpha/hummingbot/3791?planning-poker) with ZenHub @aarmoa @CrimsonJacket @james-hummingbot @petioptrv @mhrvth @vic-en

**Comment by aarmoa** [comment]
The definition of done is too abstract to serve its purpose. This task can't be estimated as it is, unless we want it to be an investigation task (but the comment in the DoD speaks about implementing, so I guess this is not an investigation task)

**Comment by phbrgnomo** [comment]
@aarmoa updated the definition of done. Added a documentation update, but we can discuss if it should be added as a separate task and/or if documentation updates should be part of tasks stories that require these updates.

**Comment by mifeng** [comment]
I agree with Abel that the description is vague, but actually specifying exactly which variables, etc to fix in the strategy is basically doing the work here. I recommend a holistic overhaul of the spot-perpetual strategy from a logic and documentation standpoint, which I would estimate 5.

**Comment by Nullably** [comment]
I suggest removing `max_funding_rate` configuration parameter for now to get the basics of the strategy right first.

**Comment by dennisocana** [comment]
This story needs to be rewritten so it's clear to engineering what's the task to be completed. Will coordinate with @Nullably and @mhrvth
